? _rombios_.c
? _rombiosl_.c
? a.out
? biossums
? rombios.s
? rombios.sym
? rombios.txt
? rombios16.bin
? rombios32.bin
? rombios32.out
? rombiosl.s
? rombiosl.sym
? rombiosl.txt
Index: BIOS-bochs-latest
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/BIOS-bochs-latest,v
retrieving revision 1.145
diff -u -r1.145 BIOS-bochs-latest
Binary files /tmp/cvsHcT9RD and BIOS-bochs-latest differ
Index: BIOS-bochs-legacy
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/BIOS-bochs-legacy,v
retrieving revision 1.9
diff -u -r1.9 BIOS-bochs-legacy
Binary files /tmp/cvszbAyyn and BIOS-bochs-legacy differ
Index: rombios.c
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios.c,v
retrieving revision 1.178
diff -u -r1.178 rombios.c
--- rombios.c	2 Apr 2007 20:35:08 -0000	1.178
+++ rombios.c	17 May 2007 13:32:36 -0000
@@ -9521,6 +9521,14 @@
   rep 
     stosd
 
+  ;; reset the memory (some boot loaders such as syslinux suppose 
+  ;; that the memory is set to zero)
+  mov edi, #0x00040000
+  mov ecx, #0x40000 / 4
+  xor eax, eax
+  rep 
+    stosd
+
   ;; return to 16 bit protected mode first
   db 0xea
   dd rombios32_10
Index: rombios.h
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios.h,v
retrieving revision 1.4
diff -u -r1.4 rombios.h
--- rombios.h	20 Feb 2007 09:36:55 -0000	1.4
+++ rombios.h	17 May 2007 13:32:36 -0000
@@ -19,7 +19,7 @@
 //  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
 
 /* define it to include QEMU specific code */
-//#define BX_QEMU
+#define BX_QEMU
 
 #ifndef LEGACY
 #  define BX_ROMBIOS32     1
Index: rombios32.c
===================================================================
RCS file: /cvsroot/bochs/bochs/bios/rombios32.c,v
retrieving revision 1.9
diff -u -r1.9 rombios32.c
--- rombios32.c	20 Feb 2007 09:36:55 -0000	1.9
+++ rombios32.c	17 May 2007 13:32:40 -0000
@@ -855,6 +855,11 @@
     int ioapic_id, i, len;
     int mp_config_table_size;
 
+#ifdef BX_QEMU
+    if (smp_cpus <= 1)
+        return;
+#endif
+
 #ifdef BX_USE_EBDA_TABLES
     mp_config_table = (uint8_t *)(ram_size - ACPI_DATA_SIZE - MPTABLE_MAX_SIZE);
 #else
